src/main/java/ch/njol/skript/test/runner/EffDebug.java

```diff
@@ -0,0 +1,141 @@
+
+import java.util.Arrays;
+import java.util.logging.Logger;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.NoDoc;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.log.ParseLogHandler;
+import ch.njol.skript.log.SkriptLogger;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+
+@NoDoc
+public class EffDebug extends Effect  {
+
+	static {
+		if (TestMode.ENABLED)
+			Skript.registerEffect(EffDebug.class, "debug [:verbose] %objects%", "debug-effect <.+>", "debug-condition <.+>");
+	}
+
+	private Expression<?> expressions;
+	private boolean debug = Skript.debug();
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (matchedPattern == 1) {
+			String string = parseResult.regexes.get(0).group();
+			ParseLogHandler logHandler = SkriptLogger.startParseLogHandler();
+			try {
+				Effect effect = Effect.parse(string, "Can't understand this effect: " + string);
+				if (effect == null) {
+					logHandler.printError();
+				} else {
+					Logger logger = Skript.getInstance().getLogger();
+					logger.info("--------------------");
+					logger.info("Parsed '" + string + "' to be Effect " + effect.getClass().getName());
+					logger.info("--------------------");
+					logHandler.printLog();
+				}
+			} finally {
+				logHandler.stop();
+			}
+			return true;
+		} else if (matchedPattern == 2) {
+			String string = parseResult.regexes.get(0).group();
+			ParseLogHandler logHandler = SkriptLogger.startParseLogHandler();
+			try {
+				Condition conditon = Condition.parse(string, "Can't understand this conditon: " + string);
+				if (conditon == null) {
+					logHandler.printError();
+				} else {
+					Logger logger = Skript.getInstance().getLogger();
+					logger.info("--------------------");
+					logger.info("Parsed '" + string + "' to be Condition " + conditon.getClass().getName());
+					logger.info("--------------------");
+					logHandler.printLog();
+				}
+			} finally {
+				logHandler.stop();
+			}
+			return true;
+		}
+		expressions = exprs[0];
+		if (LiteralUtils.canInitSafely(expressions))
+			expressions = LiteralUtils.defendExpression(expressions);
+		if (parseResult.hasTag("verbose"))
+			debug = true;
+		print();
+		return true;
+	}
+
+	@Override
+	protected void execute(Event event) {
+		if (expressions == null)
+			return;
+		print(event, debug);
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "debug " + expressions.toString(event, debug);
+	}
+
+	private void print() {
+		print(null, debug);
+	}
+
+	private void print(@Nullable Event event, boolean debug) {
+		Skript.info("--------------------");
```

Comment:
It already is a constant.

Reply:
I mean a constant field

Reply:
I don't agree with this one

Reply:
It keep the code shorter instead of typing a hypen 20 times and/or having to copy it every time you would only need to call a `SEPARATOR` so what's  you don't agree with exactly?

Reply:
Why would you need a separator for a line. This is nit picking. It does not need to be a constant.

Reply:
Both would compile in exactly the same way, so I don't think it makes a difference.